make full text search on native currency name & symbol

denciu opened the pull request:
# Summary

Closes #951 

Native currency is now included in search results on similar rules as regular token (not only when query is beginning of native currency symbol).

To test:
1. Open token search in any place (swap/bridge) on any chain (examples are on mainnet as symbol is different than name (ETH, Ether)
2. Following queries should include ETH on token list (should work even when query starts with spaces):
- eth
- er
- the
- ...etc

MilanVojnovic95 reviewed (approved):
Tested and now it works great! 👍

wixzi commented on src/components/SearchModal/CurrencySearch/CurrencySearch.component.tsx:
```diff
@@ -72,11 +72,12 @@ export const CurrencySearch = ({
-    if (nativeCurrency.symbol && nativeCurrency.symbol.toLowerCase().startsWith(s)) {
+    if (!showNativeCurrency || !nativeCurrency.symbol || !nativeCurrency.name) return filteredSortedTokens
+
+    if (new RegExp(debouncedQuery.replace(/\s/g, ''), 'gi').test(`${nativeCurrency.symbol} ${nativeCurrency.name}`)) {
       return nativeCurrency ? [nativeCurrency, ...filteredSortedTokens] : filteredSortedTokens
     }
```
```suggestion
    if (nativeCurrency  && (new RegExp(debouncedQuery.replace(/\s/g, ''), 'gi').test(`${nativeCurrency.symbol} ${nativeCurrency.name}`)) {
      return [nativeCurrency, ...filteredSortedTokens]
    }
```

Maybe in a single check :)

niemam29 commented:
Hey! I've noticed that there is some problem when searching for xdai in bridge (gnosis network)

https://user-images.githubusercontent.com/30408272/173299355-da554741-d073-4486-aa4a-0173441d63ee.mp4

denciu commented:
> Hey! I've noticed that there is some problem when searching for xdai in bridge (gnosis network)
> 
>  s.mp4

For some reason he didn't show token list if only token available was native currency. Fixed in 

> Hey! I've noticed that there is some problem when searching for xdai in bridge (gnosis network)
> 
>  s.mp4

Fixed in [669f1e6](https://github.com/levelkdev/swapr-dapp/pull/1074/commits/669f1e6fc3c80e8f109431c61d0eee0f8a084ebf)

MilanVojnovic95 commented:
Tested and it looks like the bug is no longer present, looks ok to me 👍
